src/com/lftechnology/java/training/sanish/application/model/dao/UserDao.java

```diff
@@ -0,0 +1,291 @@
+            ResultSet rs = preparedStatement.executeQuery();
+            User user = null;
+            if (rs.next()) {
+                user = new User();
+                user.setResultSetAttributes(rs);
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return user;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return null;
+    }
+
+    @Override public List<User> getAll() {
+        List<User> userList = new ArrayList<>();
+        try {
+            String query = "SELECT * FROM users WHERE isTerminated=?";
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            preparedStatement.setBoolean(1, false);
+            ResultSet rs = preparedStatement.executeQuery();
+
+            while (rs.next()) {
+                User user = new User();
+                user.setResultSetAttributes(rs);
+                userList.add(user);
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return userList;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return userList;
+    }
+
+    @Override public <V> List<User> getAll(String condition, V... parameters) {
+        List<User> userList = new ArrayList<User>();
+        try {
+            String query = "SELECT * FROM users WHERE " + condition;
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            try {
+                for (int i = 0; i < parameters.length; i++) {
+                    preparedStatement.setString(i + 1, parameters[i].toString());
+                }
+            } catch (SQLException e) {
+                LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+            while (rs.next()) {
+                User user = new User();
+                user.setResultSetAttributes(rs);
+                userList.add(user);
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return userList;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return userList;
+    }
+
+    @Override public <V> int update(String condition, String setString, V... parameters) {
+        try {
+            String query = "UPDATE users SET " + setString + " WHERE " + condition;
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1141)